### DISCUSSION TITLE
Update to new version using docker

### DISCUSSION THREAD

**Discussion by dimedime** [zwave-js/zwave-js-ui#2748] [Q&A] [answered]
Is there something like an update-button or 'signal' that a new version of zwave-js-ui is available? I have 8.02 running and saw that version 8.2.1 is already there.

But more important, how to update? I'm running the ui on a PI with docker. As a no-know, it took a lot of effort to have it running. For me the documentation was not always clear (have totally no experience with docker), but at the end the sun was shining 🥇 
I'm a little hesitating to move to the new version, did some search on the forum about updating, but couldn't find a similar question. Is it an idea to have something written on how to update?

**Comment by robertsLando** [comment, marked as answer]
Giving that there are so many different ways to install this application there are also many different ways to update it. Some of this methods like snap and Home Assistant addons can use their platform to update the package. 

In your case it's quite easy to do the update using docker. If you have a `docker-compose.yml` file and the zwave-js-ui image tag is `latest` you only have to run:

```bash
docker-compose pull  && docker-compose up -d
```

**Reply by dimedime** [reply]
Thanks for the quick reply!
I gave it a shot and got an error. It's not starting up anymore. Has it something to do with the warn-message?

WARN[0000] volume "zwave-config" already exists but was not created for project "pi". Use `external: true` to use an existing volume 
[+] Running 1/0
 â ¿ Container zwave-js-ui  Created                                          0.0s
Attaching to zwave-js-ui
zwave-js-ui  | 
zwave-js-ui  | 
zwave-js-ui  | #
zwave-js-ui  | # Fatal error in , line 0
zwave-js-ui  | # unreachable code
zwave-js-ui  | #
zwave-js-ui  | #
zwave-js-ui  | #
zwave-js-ui  | #FailureMessage Object: 0x7eb6049c
zwave-js-ui exited with code 133

**Reply by robertsLando** [reply]
That because you encountered this bug: https://github.com/zwave-js/zwave-js-ui/issues/2725

Follow instructions [here](https://github.com/zwave-js/zwave-js-ui/issues/2725#issuecomment-1275670405)

**Reply by dimedime** [reply]
Thanks! It works.
I was already trying to figure out how to get my former docker image version running. Learned a lot about docker ;-)

Again thanks!

**Reply by robertsLando** [reply]
Glad you fixed it :) You welcome
